numba/cpython/randomimpl.py

```diff
@@ -52,6 +52,20 @@ def const_int(x):
 rnd_state_ptr_t = ir.PointerType(rnd_state_t)
 
 
+def _is_valid_size(size):
+    """Check if a size has a valid type for a random function."""
+    if isinstance(size, types.Integer):
+        return True
+
+    if isinstance(size, types.UniTuple):
+        return isinstance(size.dtype, types.Integer)
+
+    if isinstance(size, types.Tuple):
+        return len(size) == 0
+
+    return False
```

Comment:
A good addition to this can be a mechanism that raises an error stating that the `size` argument type is wrong.